src/ApolloHooksMutation.re

```diff
@@ -36,72 +30,87 @@ type controlledVariantResult('a) =
   | Error(error)
   | NoData;
 
-module Make = (Config: Config) => {
-  [@bs.module "graphql-tag"] external gql: ReasonApolloTypes.gql = "default";
-
-  type mutationResult = {. "data": option(Config.t)};
-
-  [@bs.deriving abstract]
-  type options = {
-    [@bs.optional]
-    variables: Js.Json.t,
-    [@bs.optional]
-    client: ApolloClient.generatedApolloClient,
-    [@bs.optional]
-    refetchQueries,
-    [@bs.optional]
-    awaitRefetchQueries: bool,
-    [@bs.optional]
-    update: (ApolloClient.generatedApolloClient, mutationResult) => unit,
-  };
+[@bs.module "graphql-tag"] external gql: ReasonApolloTypes.gql = "default";
+
+type mutationResult('a) = {. "data": option('a)};
+
+[@bs.deriving abstract]
+type options('a) = {
+  [@bs.optional]
+  variables: Js.Json.t,
+  [@bs.optional]
+  mutation: option(ReasonApolloTypes.queryString),
+  [@bs.optional]
+  client: ApolloClient.generatedApolloClient,
+  [@bs.optional]
+  refetchQueries,
+  [@bs.optional]
+  awaitRefetchQueries: bool,
+  [@bs.optional]
+  update: (ApolloClient.generatedApolloClient, mutationResult('a)) => unit,
+};
 
-  type jsResult = {
-    .
-    "data": Js.Nullable.t(Js.Json.t),
-    "loading": bool,
-    "called": bool,
-    "error": Js.Nullable.t(error),
-  };
+type jsResult = {
+  .
+  "data": Js.Nullable.t(Js.Json.t),
+  "loading": bool,
+  "called": bool,
+  "error": Js.Nullable.t(error),
+};
 
-  type jsMutate = (. options) => Js.Promise.t(jsResult);
-  type mutation =
+type jsMutate('a) = (. options('a)) => Js.Promise.t(jsResult);
+type mutation('a) =
+  (
+    ~variables: Js.Json.t=?,
+    ~client: ApolloClient.generatedApolloClient=?,
+    ~refetchQueries: refetchQueries=?,
+    ~awaitRefetchQueries: bool=?,
+    unit
+  ) =>
+  Js.Promise.t(controlledVariantResult('a));
+
+[@bs.module "@apollo/react-hooks"]
+external useMutationJs:
+  (. ReasonApolloTypes.queryString, options('a)) => (jsMutate('a), jsResult) =
+  "useMutation";
+
+exception Error(string);
+
+let useMutation:
+  type t.
     (
-      ~variables: Js.Json.t=?,
       ~client: ApolloClient.generatedApolloClient=?,
+      ~variables: Js.Json.t=?,
       ~refetchQueries: refetchQueries=?,
       ~awaitRefetchQueries: bool=?,
-      unit
+      ~update: (ApolloClient.generatedApolloClient, mutationResult('a)) =>
```

Comment:
Does anyone know whether this `'a` in `mutationResult('a)`  is the same as `t`?

Reply:
It was the same in the old version, though I really doubt it will work correctly since `update` doesn't parse the response with `parse`, but we will have to fix in another PR 😄

Reply:
I read a bit fast, but I think the `'a` and the old `t` are kinda having the same job